internal/infrastructure/kubernetes/proxy/resource.go

```diff
@@ -158,10 +158,15 @@ func expectedProxyContainers(infra *ir.ProxyInfra, deploymentConfig *egcfgv1a1.K
 			ReadinessProbe: &corev1.Probe{
 				ProbeHandler: corev1.ProbeHandler{
 					HTTPGet: &corev1.HTTPGetAction{
-						Path: bootstrap.EnvoyReadinessPath,
-						Port: intstr.IntOrString{Type: intstr.Int, IntVal: bootstrap.EnvoyReadinessPort},
+						Path:   bootstrap.EnvoyReadinessPath,
+						Port:   intstr.IntOrString{Type: intstr.Int, IntVal: bootstrap.EnvoyReadinessPort},
+						Scheme: corev1.URISchemeHTTP,
 					},
 				},
+				TimeoutSeconds:   1,
```

Comment:
any reason why this is being set, can this be moved into a separate PR, tia

Reply:
this belongs to this fix.

the `TimeoutSecounds` will fill the default value, and compare the `current` from k8s and `deployment` from code, it never equal forever will fire current bug : more terminating pod and deployment generation will increment all the time

Reply:
![image](https://github.com/envoyproxy/gateway/assets/2174082/951182f9-472d-4eae-8299-d25d7b5aa244)